R/match_on_ltcs.R

```diff
@@ -0,0 +1,171 @@
+#' Match on LTC changed_dob and dates of LTC incidence (based on hospital incidence only)
+#'
+#' @description Match on LTC changed_dob and dates of LTC incidence (based on hospital incidence only)
+#'
+#' @param
+#'
+#' @return
+#' @export
+#'
+#' @examples
+match_on_ltcs <- function() {
+  # Get temp file
+  temp <-
+    haven::read_sav(
+      "/conf/sourcedev/Source_Linkage_File_Updates/1920/temp-source-episode-file-4-1920-testing.zsav",
+      # "~/temp-source-episode-file-4-1920.zsav",
+      # n_max = n_maxx
+      n_max = 1e6
+    )
+
+  # Match on LTC lookup
+  matched <- dplyr::left_join(temp,
+    readr::read_rds(get_ltcs_path("1920")),
+    by = "chi",
+    suffix = c("", "_ltc")
+  ) %>%
+    # Replace any NA values with 0 for the LTC flags
+    dplyr::mutate(dplyr::across(arth:digestive, ~ tidyr::replace_na(., 0)))
+
+  # Checking and changing DOB and age
+  test <- matched %>%
+    dplyr::filter(!is_missing(chi)) %>%
+    dplyr::mutate(
+      # Create a dob in the previous century from the chi number
+      chi_dob_min = phsmethods::dob_from_chi(
+        chi_number = chi,
+        chi_check = FALSE,
+        min_date = lubridate::ymd("1900-01-01"),
+        max_date = lubridate::ymd("2000-01-01")
+      ),
+      # Create a dob in the current century from chi (will return NA if in the future)
+      chi_dob_max = phsmethods::dob_from_chi(
+        chi_number = chi,
+        chi_check = FALSE,
+        min_date = lubridate::ymd("2000-01-01")
+      ),
+      # Compute two ages for each chi, the maximum and minimum it could be
+      chi_age_max = compute_mid_year_age("1920", chi_dob_min),
+      chi_age_min = compute_mid_year_age("1920", chi_dob_max),
+      # Mutate age based on different scenarios
+      age = dplyr::case_when(
+        # Case when the dob is already valid
+        .data$dob == chi_dob_min ~ chi_age_max,
+        .data$dob == chi_dob_max ~ chi_age_min,
+        # If one option for dob isn't there, use the other
+        is.na(chi_dob_min) & !is.na(chi_dob_max) ~ chi_age_min,
+        !is.na(chi_dob_min) & is.na(chi_dob_max) ~ chi_age_max,
+        # If the age would be negative, use the older
+        chi_age_min < 0 ~ chi_age_max,
```

Comment:
I don't think this would ever be the case now we're using `phsmethods::dob_from_chi`. Could you double check please but I think this condition can be removed.
```suggestion
```

Reply:
Yes, `dob_from_chi` does.